Tribler/Core/Modules/restapi/my_channel_endpoint.py

```diff
@@ -88,64 +103,154 @@ def render_PUT(self, request):
+            channel_id = self.session.create_channel(parameters['name'][0], description, mode)
         except DuplicateChannelNameError as ex:
-            request.setResponseCode(http.INTERNAL_SERVER_ERROR)
-            request.setHeader('Content-Type', 'text/json')
-            return json.dumps({
-                u"error": {
-                    u"handled": True,
-                    u"code": ex.__class__.__name__,
-                    u"message": ex.message
-                }
-            })
+            return MyChannelBaseEndpoint.return_500(self, request, ex)
 
-        request.setHeader('Content-Type', 'text/json')
         return json.dumps({"added": channel_id})
 
 
 class MyChannelTorrentsEndpoint(MyChannelBaseEndpoint):
     """
-    Return the torrents in your channel. For each torrent item, the infohash, name and timestamp added is included.
-    This endpoint returns a 404 HTTP response if you have not created a channel (yet).
-
-    Example response:
-    {
-        "torrents": [{
-            "name": "ubuntu-15.04.iso",
-            "added": 1461840601,
-            "infohash": "e940a7a57294e4c98f62514b32611e38181b6cae"
-        }, ...]
-    }
+    This end is responsible for handling requests regarding torrents in your channel.
     """
 
+    def getChild(self, path, request):
+        return MyChannelModifyTorrentsEndpoint(self.session, path)
+
     def render_GET(self, request):
+        """
+        Return the torrents in your channel. For each torrent item, the infohash, name and timestamp added is included.
+        This endpoint returns a 404 HTTP response if you have not created a channel (yet).
+
+        Example response:
+        {
+            "torrents": [{
+                "name": "ubuntu-15.04.iso",
+                "added": 1461840601,
+                "infohash": "e940a7a57294e4c98f62514b32611e38181b6cae"
+            }, ...]
+        }
+        """
         my_channel_id = self.channel_db_handler.getMyChannelId()
         if my_channel_id is None:
             return MyChannelBaseEndpoint.return_404(request)
 
         req_columns = ['ChannelTorrents.name', 'infohash', 'ChannelTorrents.inserted']
         torrents = self.channel_db_handler.getTorrentsFromChannelId(my_channel_id, True, req_columns)
 
-        request.setHeader('Content-Type', 'text/json')
         torrent_list = []
         for torrent in torrents:
             torrent_list.append({'name': torrent[0], 'infohash': torrent[1].encode('hex'), 'added': torrent[2]})
         return json.dumps({'torrents': torrent_list})
 
+    def render_PUT(self, request):
+        """
+        Add a torrent file to your own channel.Returns error 500 if something is wrong with the torrent file
+        and DuplicateTorrentFileError if already added to your channel.
+
+        Example request:
+        {
+            "torrent": "base64 encoded string of torrent file contents",
+            "description" (optional): "A video of my cat" (default: empty)
```

Comment:
That's interesting. @synctext what do you say? It would make adding torrents from the command line (via curl for instance) simpler.